pkg/aggregator/check_sampler.go

```diff
@@ -90,12 +90,16 @@ func (cs *CheckSampler) addBucket(bucket *metrics.HistogramBucket) {
 	if bucket.Monotonic {
 		lastBucketValue, bucketFound := cs.lastBucketValue[contextKey]
 		rawValue := bucket.Value
-		if bucketFound {
-			cs.lastSeenBucket[contextKey] = time.Now()
-			bucket.Value = rawValue - lastBucketValue
-		}
-		cs.lastBucketValue[contextKey] = rawValue
+
 		cs.lastSeenBucket[contextKey] = time.Now()
+		cs.lastBucketValue[contextKey] = rawValue
+
+		// Return early so we don't report the first raw value instead of the delta which will cause spikes
+		if !bucketFound {
+			return
```

Comment:
If I understand this properly we will wait to only send deltas of the bucket to avoid bump in case of agent restart, my main concern is about the case where the monitored app has been running for a while (exposing lots of points in the histogram) and we start collecting it with the agent, all this history will not be taken into consideration on the first collection due to this change?

Reply:
This is true - however on the flip side, if we have already processed all of this historical data and then the agent restarts for some reason we will have a massive spike that represents data we have already processed in the past. 

I guess the tradeoff is which is more important to us? CC @olivielpeau

Reply:
yes it's definitely a tradeoff, it would be nice to apply this only to agent restarts and not on the first collecting but I guess it could be tricky to identify starts from restarts 🤔

Reply:
in the nominal case where the agent doesn't restart people might be confused why the histogram doesn't reflect the counts/data that is exposed in their prometheus endpoint (the values never exactly matched because of interpolation but the count were supposed to match which will not be the case anymore if we skip the first data point)

Reply:
Interesting, this is actually a very similar problem that we addressed on simple Openmetrics Counters sent as Datadog Count metrics: https://github.com/DataDog/datadog-agent/pull/6663

Overall:

* when the agent restarts or when it's the first time the check runs against an openmetrics endpoint, there's no good reliable and simple way at the moment to tell if the monotonic values we get from the openmetrics endpoints are relevant or not, so I it makes sense to not send these raw values. The present PR fixes this case.
* but, once the check has run more than once against an openmetrics endpoint, if there's a new Openmetrics histogram (new = new metric name or new tag set) that's exposed by the endpoint, then the Agent can tell that the raw values of this histogram can be sent as-is (since the previous values of the histogram are `0`). The present PR actually breaks this case: without this change, new metrics that appear on an already-scraped Openmetrics histogram would have their first values sent correctly, whereas with this change, these values are discarded.

So the conclusion is that I think we should hold off of merging this PR, and plan to have a mechanism similar to https://github.com/DataDog/datadog-agent/pull/6663 for histogram buckets.

(let me know how that sounds)